pkg/services/authn/authnimpl/sync/oauth_token_sync.go

```diff
@@ -113,15 +125,49 @@ func (s *OAuthTokenSync) SyncOauthTokenHook(ctx context.Context, identity *authn
 
 const maxOAuthTokenCacheTTL = 10 * time.Minute
 
-func getOAuthTokenCacheTTL(t time.Time) time.Duration {
-	if t.IsZero() {
+func getOAuthTokenCacheTTL(accessTokenExpiry, idTokenExpiry time.Time) time.Duration {
+	if accessTokenExpiry.IsZero() {
 		return maxOAuthTokenCacheTTL
 	}
 
-	ttl := time.Until(t)
-	if ttl > maxOAuthTokenCacheTTL {
-		return maxOAuthTokenCacheTTL
+	accessTokenTTL := time.Until(accessTokenExpiry)
+	if accessTokenTTL > maxOAuthTokenCacheTTL {
+		accessTokenTTL = maxOAuthTokenCacheTTL
+	}
+
+	if idTokenExpiry.IsZero() {
+		return accessTokenTTL
+	}
+
+	idTokenTTL := time.Until(idTokenExpiry)
+	if idTokenTTL > maxOAuthTokenCacheTTL {
+		idTokenTTL = maxOAuthTokenCacheTTL
+	}
+
+	if accessTokenTTL <= idTokenTTL {
+		return accessTokenTTL
+	}
+  return idTokenTTL
```

Comment:
This was my mistake with the suggestion. This change will help you with the linter check.
```suggestion
	return idTokenTTL
```

Reply:
Hello @linoman, I refactored this function based on @mgyongyosi's feedback which should have resolved this issue.